app/models/user.rb

```diff
@@ -3,4 +3,17 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  has_one_attached :avatar
+
+  validate :avatar_content_type, if: :was_attached?
+
+  def avatar_content_type
+    extension = ['image/jpeg', 'image/jpg', 'image/png', 'image/gif']
```

Comment:
```suggestion
    extensions = ['image/jpeg', 'image/jpg', 'image/png', 'image/gif']
```

配列は複数形にすることが多いですね。
あと、毎回同じ値になるので定数化する方がいいと思います。

Reply:
確かに配列の中身は変更しないため定数の方がよいですね。
定数にして、配列名を複数形にしました。